streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValues.java

```diff
@@ -36,29 +36,29 @@
 
 public class KStreamTransformValues<K, V, R> implements ProcessorSupplier<K, V> {
 
-    private final ValueTransformerSupplier<V, R> valueTransformerSupplier;
+    private final ValueTransformerWithKeySupplier<K, V, R> valueTransformerWithKeySupplier;
 
-    public KStreamTransformValues(ValueTransformerSupplier<V, R> valueTransformerSupplier) {
-        this.valueTransformerSupplier = valueTransformerSupplier;
+    public KStreamTransformValues(ValueTransformerWithKeySupplier<K, V, R> valueTransformerWithKeySupplier) {
```

Comment:
nit: `final`
I know it wasn't before, but let's stick to making params as final